wallet_presentation/src/commonMain/kotlin/net/thechance/mena/wallet/presentation/screen/transaction_details/TransactionDetailsMapper.kt

```diff
@@ -20,46 +20,71 @@ import net.thechance.mena.wallet.presentation.utils.formatLocalDateTime
 import kotlin.uuid.ExperimentalUuidApi
 
 fun Transaction.toUi() = TransactionDetailsScreenState.TransactionDetailsUiState(
-    id = Constants.ID_PREFIX + id.toString().substring(0,6),
-    amount = amount.toString().replace(".",","),
-    date = formatLocalDateTime(date = createdAt, outputFormat = "dd MMM yyyy, h:mm a"),
-    userName = when (type) {
-        TransactionType.SENT -> senderName
-        TransactionType.RECEIVED -> receiverName
-        TransactionType.ONLINE_PURCHASE -> senderName
-    },
-    otherParty = when (type) {
-        TransactionType.SENT -> receiverName
-        TransactionType.RECEIVED -> senderName
-        TransactionType.ONLINE_PURCHASE -> receiverName
-    },
-    transactionType = when (type) {
-        TransactionType.SENT -> TransactionTypeUiState.SENT
-        TransactionType.RECEIVED -> TransactionTypeUiState.RECEIVED
-        TransactionType.ONLINE_PURCHASE -> TransactionTypeUiState.ONLINE_PURCHASE
-    },
-    transactionStatus = when (status) {
-        TransactionStatus.SUCCESS -> TransactionStatusUiState.SUCCESS
-        TransactionStatus.FAILED -> TransactionStatusUiState.FAILED
-    },
-    userInfo = when (type) {
-        TransactionType.SENT, TransactionType.ONLINE_PURCHASE -> Res.string.from
-        TransactionType.RECEIVED -> Res.string.to
-    },
-    typeContent = when (type) {
-        TransactionType.SENT, TransactionType.RECEIVED -> Res.string.transfer
-        TransactionType.ONLINE_PURCHASE -> Res.string.purchase
-    },
-    otherPartyTitle = when (type) {
-        TransactionType.SENT -> Res.string.to
-        TransactionType.RECEIVED -> Res.string.from
-        TransactionType.ONLINE_PURCHASE -> Res.string.receiver
-    }
+    id = transactionId(),
```

Comment:
what about adding `this` before those functions for more readability?
```suggestion
    id = this.transactionId(),
```